script.js

```diff
@@ -0,0 +1,79 @@
+/* add todo*/
+const todoInput=document.getElementById("todo-input");
+const addTodoButton=document.getElementById("todo-button");
+const doingTodoList=document.getElementById("doing");
+const doneTodoList=document.getElementById("done");
+let todoArray=[];
+const doingCount=document.getElementById("doing-count");
+const doneCount=document.getElementById("done-count");
+
+/*create todo li*/
+function createTodoList(todoObj){
+    const todoListItem=document.createElement("li");
+    const deleteImg=document.createElement("img");
+    todoListItem.innerText="🐈 "+todoObj.text;
+    todoListItem.id=todoObj.id;
+    todoListItem.addEventListener("click", toggleTodo);
+    deleteImg.src="./img/bin.png";
+    deleteImg.addEventListener("click", deleteTodo);
+    todoListItem.appendChild(deleteImg);
+    todoObj.type? doingTodoList.appendChild(todoListItem):doneTodoList.appendChild(todoListItem);
+}
+
+/*click add todo button*/
+function onClickAddTodo(e){
+    e.preventDefault();
+    if (todoInput.value){
+        const todoObj={
+        "text":todoInput.value,
+        "id":Date.now(),
+        "type": true,
+        }
+        createTodoList(todoObj);
+        todoArray.push(todoObj);
+        saveTodo();
+    }
+    todoInput.value="";
+}
+addTodoButton.addEventListener("click", onClickAddTodo);
+
+/*doing->done, done->doing*/
+function toggleTodo(e){
+    todoArray.forEach(todo=>{
+        if (todo.id===parseInt(e.target.id)){
+            const li=e.target; 
+            li.remove();
+            //type을 반전
+            todo.type=!todo.type;
+            createTodoList(todo);
+            saveTodo();
+        }
+    })
```

Comment:
javascript array method를 사용하신 점이 인상깊네요!

`.find()` 메소드를 사용해보는 것도 좋을 것 같습니다.

```suggestion
const targetTodoLiElement = e.target;
const targetTodoId = parseInt(targetTodoLiElement.id);
const targetTodo = todoArray.find((todo) => todo.id === targetTodoId);
targetTodoLiElement.remove();
todo.type = !todo.type;
createTodoList(todo);
saveTodo()
```

또 이런 식으로 변수 이름을 명확하게 지어서 코드의 가독성을 올려보는 것도 좋을 것 같습니다.

Reply:
이 부분 작성하면서도 아 뭔가 아쉬운데...?라는 느낌이 있었는데 find를 쓰면 훨씬 깔끔해지네요! 😮